strawberryfields/plot.py

```diff
@@ -0,0 +1,152 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+This module provides tools to visualize the state in various interactive
+ways using Plot.ly
+"""
+from copy import deepcopy
+
+import numpy as np
+
+plotly_error = (
+    "Plot.ly required for using this function. It can be installed using pip install "
+    "plotly or visiting https://plot.ly/python/getting-started/#installation"
+)
+
+try:
+    import plotly.io as pio
+except ImportError as e:
+    raise(plotly_error) from e
+
+
+def plot_wigner(state, p_axis, q_axis, renderer="browser", wire=0, dq=0.1):
+    """Plot the Wigner function with Plot.ly
+
+    Args:
+        state (:class:`.BaseState`): the state used for plotting
+        p_axis (int): Range of the discretized :math:`p` quadrature values.
+            E.g., ranging from ``-p_axis`` to ``p_axis``.
+        q_axis (int): Range of the discretized :math:`x` quadrature values.
+            E.g., ranging from ``-q_axis`` to ``q_axis``.
+        renderer (string): the renderer for plotting with Plot.ly
+        wire (int): wire used to calculate the reduced Wigner function
+        dq (float): stepsize along the p_axis and q_axis
+    """
+    xvec = np.arange(-q_axis, q_axis, dq)
+    pvec = np.arange(-p_axis, p_axis, dq)
```

Comment:
There could be a case where the user wants to pass `xvec` or `pvec` directly. E.g., if the state to be plotted is far from the origin, you wouldn't want the axes to be symmetric around 0.

Perhaps it's better to just pass `xvec` and `pvec`, like you do with matplotlib, and like the `generate_wigner_chart` function.